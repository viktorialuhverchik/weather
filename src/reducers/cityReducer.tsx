import { SELECT_CITY } from '../types';

const initialState = {
    city: "London"
}

export const cityReducer = (state = initialState, action: any) => {
    switch (action.type) {
        case SELECT_CITY:
            return {...state, city: state.city.concat(action.payload)};
        default: return state;
    }
} 