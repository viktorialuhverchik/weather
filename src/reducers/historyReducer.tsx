import { CREATE_HISTORY } from '../types';

const initialState: any = {
    history: []
};

export const historyReducer = (state = initialState, action: any) => {
    switch (action.type) {
        case CREATE_HISTORY:
            return {...state, history: action.payload};
        default: 
            return state;
    }
};