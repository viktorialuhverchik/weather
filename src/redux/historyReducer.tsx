import { CREATE_HISTORY, UPDATE_HISTORY, DELETE_HISTORY } from './types';

export const initialState: any = [];

export const historyReducer = (state = initialState, action: any) => {
    switch (action.type) {
        case CREATE_HISTORY:
            return action.history;
        case UPDATE_HISTORY:
            let history = [...state];
            if(history.length === 10) {
                history.pop();
            }
            let newHistory = history.filter((item: any) => item !== action.history);
            newHistory.unshift(action.history);
            localStorage.setItem("history", JSON.stringify(newHistory));
            return newHistory;
        case DELETE_HISTORY:
            const updatedHistory = state.filter((item: any) => item !== action.history );
            localStorage.setItem("history", JSON.stringify(updatedHistory));
            return updatedHistory;
        default: 
            return state;
    }
};