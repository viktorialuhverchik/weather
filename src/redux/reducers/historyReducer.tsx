import { CREATE_HISTORY, UPDATE_HISTORY, DELETE_HISTORY } from '../types';

const initialState: any = [];

export const historyReducer = (state = initialState, action: any) => {
    switch (action.type) {
        case CREATE_HISTORY:
            return action.history;
        case UPDATE_HISTORY:
            let history = [...state];
            let newHistory = history.filter((item: any) => item !== action.history);
            if(newHistory.length === 10) {
                newHistory.pop();
            }
            newHistory.unshift(action.history);
            localStorage.setItem("history", JSON.stringify(newHistory));
            return newHistory;
        case DELETE_HISTORY:
            const updatedHistory = state.filter((item: any) => item !== action.history );
            localStorage.setItem("history", JSON.stringify(updatedHistory));
            return updatedHistory;
        default: 
            return state;
    }
};