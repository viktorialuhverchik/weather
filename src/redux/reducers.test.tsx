import { historyReducer, initialState } from './historyReducer';
import { cityReducer, initialState as initState  } from './cityReducer';

describe('history reducer', () => {
    it('should return the initial state', () => {
        expect(historyReducer(undefined, {})).toEqual([]);
    });
    
    it('should handle CREATE_HISTORY', () => {
        expect(historyReducer([], 
            {
            type: 'CREATE_HISTORY',
            history: ["Test"]
            }
            ))
            .toEqual(["Test"]);
    });

    it('should handle UPDATE_HISTORY', () => {
        expect(historyReducer(
            [],
            {
                type: 'UPDATE_HISTORY',
                history: "London"
            }
            ))
            .toEqual(["London"]);

        expect(historyReducer(
            ["Paris"],
            {
                type: 'UPDATE_HISTORY',
                history: "London"
            }
            ))
            .toEqual(["London", "Paris"]);
    });

    it('should handle DELETE_HISTORY', () => {
        expect(historyReducer(
            ["London", "Paris"],
            {
                type: 'DELETE_HISTORY',
                history: "Paris"
            }
            ))
            .toEqual(["London"]);
    });
});

describe('city reducer', () => {
    it('should return the initial state', () => {
        expect(cityReducer(undefined, 'London')).toMatchSnapshot()
    });
    
    it('should handle SELECT_CITY', () => {
        expect(cityReducer(initState, {type: 'SELECT_CITY'})).toMatchSnapshot()
    });
});

