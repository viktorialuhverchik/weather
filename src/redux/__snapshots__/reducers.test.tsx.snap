// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`city reducer should handle SELECT_CITY 1`] = `
Object {
  "name": undefined,
}
`;

exports[`city reducer should return the initial state 1`] = `
Object {
  "name": "London",
}
`;
